### PR TITLE
Message value unit tests 

### PR DESCRIPTION
**Pull request by alexander871015**
Implementation for unit tests for message value's public api.

### CODE REVIEW

File: proto_parser/src/message_value_test.cc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "message_value.h"
+#include "gmock/gmock.h"
+
+namespace wireless_android_play_analytics{
+ 
+class MessageValueTest : public ::testing::Test {
+ protected:
+  MessageValueTest()
+    : message_(MessageValue("test", 0, 0)) { }
```
Comment (sarahchen-google) [inline proto_parser/src/message_value_test.cc:25]: Move the message_ initialization into SetUp()

Comment (sarahchen-google) [inline proto_parser/src/message_value_test.cc:25]: Add /*xxx*/ for the mystery values 0, 1, 2

File: proto_parser/src/message_value_test.cc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+TEST_F(MessageValueTest, ProtoValueGetterTests) {
+  EXPECT_EQ(message_.GetName(), "test");
+  EXPECT_EQ(message_.GetCommentAboveField().size(), 0);
+  EXPECT_EQ(message_.GetCommentBehindField(), "");
+  EXPECT_EQ(message_.GetLineNumber(), 0);
+  EXPECT_EQ(message_.GetIndentCount(), 0);
+}
+
+MATCHER_P(ProtoValuePtrIsEqual, element, "") {
```
Comment (sarahchen-google) [inline proto_parser/src/message_value_test.cc:53]: What does "" mean here?

File: proto_parser/src/message_value_test.cc [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    field_2_copy = absl::make_unique<MessageValue>("temp2", 0, 2);
+  }
+
+  MessageValue message_;
+  std::unique_ptr<ProtoValue> field_0_;
+  std::unique_ptr<ProtoValue> field_1_;
+  std::unique_ptr<ProtoValue> field_2_;
+  std::unique_ptr<ProtoValue> field_0_copy;
+  std::unique_ptr<ProtoValue> field_1_copy;
+  std::unique_ptr<ProtoValue> field_2_copy;
```
Comment (sarahchen-google) [inline proto_parser/src/message_value_test.cc:42]: You can move these variables into each test.

File: proto_parser/src/message_value_test.cc (hunk 2 above)
Comment (alexander871015) [reply proto_parser/src/message_value_test.cc:53]: It's the description of the matcher

**Pull request merged**